docs: multithread linting in v9.34.0

fasttime opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [OpenJS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/.github/blob/master/CONTRIBUTING.md).

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Update or create tests
    - If performance-related, include a benchmark
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What is the purpose of this pull request?

Added a blog post announcing multithread linting in ESLint v9.34.0

#### What changes did you make? (Give an overview)

#### Related Issues

eslint/eslint#19794

<!-- include tags like "fixes #123" or "refs #123" -->

#### Is there anything you'd like reviewers to focus on?

fasttime commented on src/content/blog/2025-08-22-multithread-linting.md:
```diff
@@ -0,0 +1,116 @@
+
+Multithreading is enabled through the new `--concurrency` flag. You can set it in a few different ways:
+
+- **A positive integer** (e.g., `4`) tells ESLint the maximum number of threads to use. It will never spawn more threads than there are files to lint.  
+- **`auto`** lets ESLint decide the optimal number of threads based on your CPU and file count.  
+- **`off`** (the default) disables multithreading, equivalent to `--concurrency=1`.
```
```suggestion
* **A positive integer** (e.g., `4`) tells ESLint the maximum number of threads to use. It will never spawn more threads than there are files to lint.  
* **`auto`** lets ESLint decide the optimal number of threads based on your CPU and file count.  
* **`off`** (the default) disables multithreading, equivalent to `--concurrency=1`.
```

fasttime commented on src/content/blog/2025-08-22-multithread-linting.md:
```diff
@@ -0,0 +1,116 @@
+  - Concurrency
+  - Multithreading
+  - Performance
+draft: true
```
```suggestion
```

fasttime commented on src/content/blog/2025-08-22-multithread-linting.md:
```diff
@@ -0,0 +1,116 @@
+authors:
+  - fasttime
+categories:
+  - Storytime
```
```suggestion
  - Announcements
```

nzakas commented on src/content/blog/2025-08-22-multithread-linting.md:
```diff
@@ -0,0 +1,116 @@
+
+## Introduction
+
+With ESLint v9.34.0, linting can now happen in true parallel. By spawning worker threads, ESLint can process multiple files at the same time, dramatically reducing lint times for large projects.  
```
I think it would be nice to start off with how old the issue is. :smile: It's a nice story to say that an issue that was opened years ago has finally been resolved.

nzakas commented on src/content/blog/2025-08-22-multithread-linting.md:
```diff
@@ -0,0 +1,116 @@
+
+Technical hurdles soon followed. Chief among them was the requirement that all options passed to worker threads be cloneable, a restriction that ruled out certain common patterns such as passing functions or complex plugin objects directly. This limitation ultimately inspired the creation of options modules, which sidestep the cloneability problem entirely by letting each worker import the same m ...
+
+Bringing multithread linting to life was a collaborative effort, shaped by the ideas, feedback, and testing of both the ESLint team and the wider community. We are thankful to everyone involved in the process of making this concept become a reality.
```
I'd move "History" and "Challenges" to the top to create a better story arch.

nzakas reviewed (commented):
This looks like a good start. I left some notes on how to clean up the telling of this story.

Also, let's be sure to mention that concurrency=auto may not be the best setting for everyone and that people should try different settings to find the one that yields the best result. Maybe mention that it seems half of the number of *physical* cores seems to be the sweet spot for most folks, but there's no way to tell the number of physical cores from within Node.js.

fasttime commented:
Thanks for the suggestions. I updated the intro paragraph and moved the History and Challenges sections up to the top. I also added a few more points to the Challenges section and broke it up into subsections. The last paragraph from that section felt like a better fit for the Way Forward section, so I moved it there, then I moved the whole Way Forward part to the end. I also reworded the How It Works section and put it before the CLI usage and Node.js API usage sections.

fasttime commented:
Added a new section to explain the limitations of `auto` concurrency in 67ae945402465f8f436ae7ffa85b2e49a38899d4.

nzakas commented on src/content/blog/2025-08-22-multithread-linting.md:
```diff
@@ -13,11 +13,37 @@ categories:
-On machines with multiple CPU cores and fast storage, this change can make a noticeable difference — especially when you're linting hundreds or thousands of files.
+The idea of parallelizing the linting process has been around for well over a decade, long before Node.js offered built-in worker threads. Early discussions wrestled with fundamental questions: which tools should be used for parallelization, how far should the scope extend, and what architectural changes would be required to make it possible. There were also concerns about the potential downsides ...
+
+In the meantime, other tools forged ahead. Test runners like Ava, Mocha, and Jest began to ship with built-in parallel execution, and inventive developers found ways to run ESLint in parallel using wrappers and external scripts. These experiments proved that parallel linting could be valuable, but they also highlighted the limitations of bolting it on from the outside. Ultimately, the only way to ...
```
Let's add links to these other projects.

nzakas reviewed (commented):
This is looking really good, I just left a few places where it would be helpful to add links and details.
